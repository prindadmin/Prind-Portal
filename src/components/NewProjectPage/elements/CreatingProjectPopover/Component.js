import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
