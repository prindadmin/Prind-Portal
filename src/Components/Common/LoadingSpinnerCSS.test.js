
import { shallow } from 'enzyme';

import Component from './LoadingSpinnerCSS'

it('Loading Spinner - Should render at default size', () => {
  const props = {}
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('Loading Spinner - Should render at smaller size', () => {
  const props = {
    size: 25
  }
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});
