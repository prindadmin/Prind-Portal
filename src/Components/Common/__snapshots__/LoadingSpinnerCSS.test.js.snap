// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Spinner - Should render at default size 1`] = `
<Fragment>
  <div
    className="lds-ring"
    style={
      Object {
        "borderWidth": 12.5,
        "height": 100,
        "margin": 12.5,
        "width": 100,
      }
    }
  >
    <div
      style={
        Object {
          "borderWidth": 12.5,
          "height": 100,
          "margin": 12.5,
          "width": 100,
        }
      }
    />
    <div
      style={
        Object {
          "borderWidth": 12.5,
          "height": 100,
          "margin": 12.5,
          "width": 100,
        }
      }
    />
    <div
      style={
        Object {
          "borderWidth": 12.5,
          "height": 100,
          "margin": 12.5,
          "width": 100,
        }
      }
    />
    <div
      style={
        Object {
          "borderWidth": 12.5,
          "height": 100,
          "margin": 12.5,
          "width": 100,
        }
      }
    />
  </div>
</Fragment>
`;

exports[`Loading Spinner - Should render at smaller size 1`] = `
<Fragment>
  <div
    className="lds-ring"
    style={
      Object {
        "borderWidth": 3.125,
        "height": 25,
        "margin": 3.125,
        "width": 25,
      }
    }
  >
    <div
      style={
        Object {
          "borderWidth": 3.125,
          "height": 25,
          "margin": 3.125,
          "width": 25,
        }
      }
    />
    <div
      style={
        Object {
          "borderWidth": 3.125,
          "height": 25,
          "margin": 3.125,
          "width": 25,
        }
      }
    />
    <div
      style={
        Object {
          "borderWidth": 3.125,
          "height": 25,
          "margin": 3.125,
          "width": 25,
        }
      }
    />
    <div
      style={
        Object {
          "borderWidth": 3.125,
          "height": 25,
          "margin": 3.125,
          "width": 25,
        }
      }
    />
  </div>
</Fragment>
`;
